solver/llbsolver/solver.go

```diff
@@ -173,12 +175,20 @@ func (s *Solver) Solve(ctx context.Context, id string, sessionID string, req fro
 			}
 			inp.Ref = workerRef.ImmutableRef
 
-			dt, err := inlineCache(ctx, exp.CacheExporter, r, session.NewGroup(sessionID))
+			dtbi, err := buildinfo.Merge(ctx, res.BuildInfo(), inp.Metadata[exptypes.ExporterImageConfigKey])
```

Comment:
why can't this be just inside the imagewriter exporter?

Reply:
we need the merged result for the Exporter response too (metadata).

Reply:
but metadata is also returned by the exporter

Reply:
I guess I done that in the solver to avoid changing the ExporterInstance.Export signature because ResultProxy is not passed to it. Do you see any downside about it?

Reply:
Not a strict requirement but thought it could be cleaner. You can test in a follow-up and see if it improves.